### ISSUE TITLE
File Finder: improve usability

### ISSUE THREAD

**Issue by jansol** [zed-industries/zed#10201]
### Check for existing issues

- [X] Completed

### Describe the feature

The file finder is very nice! Saves me a lot of scrolling through hundreds of similarly prefixed files. Especially since it supports adding :1234 to the end to immediately jump to a given number it comes in handy when going through stack traces.

However it would be more reassuring that it actually decided on the correct file if it showed a "phantom" string in the search field showing the full currently selected result (darkened, the actually entered characters in normal color, same logic as is used for highlighting the search term in the list below the field).

I'm not sure if this phantom string should be shown at all times (if the phantom part uses the "muted" color it's probably fine to always show), but once a `:` is entered to add the line number it would be nice to see the full path in the field as visual confirmation.

**Comment by maxdeviant** [comment]
@jansol I'm not sure I totally follow on what you mean by "phantom string" here. Would you be able to include some rough mockups to help visualize what it is you'd like to see?

**Comment by jansol** [comment]
I would like to see what file it would pick if I pressed enter right now. With the parts I actually typed highlighted, just like in the file listing below the text box:

![Image](https://github.com/user-attachments/assets/f53230a8-cd64-4135-b19a-72afbd6954eb)

But I would like to see it directly *in the input field* similar to to how it shows a hint string before typing anything

![Image](https://github.com/user-attachments/assets/a83310bd-722a-4fba-95a3-9a3967185db0)

So basically:

![Image](https://github.com/user-attachments/assets/06813d8a-fd7b-44d6-a655-461c09bfea6f)

(with the list also shown below *in case* the first match is not what I wanted)
